quickjs.c

```diff
@@ -8007,40 +8007,51 @@ JSValue JS_GetPropertyUint32(JSContext *ctx, JSValue this_obj,
     return JS_GetPropertyValue(ctx, this_obj, js_uint32(idx));
 }
 
+/* Check if an object has a property. Return value:
+   -1 for exception,
+   TRUE if property exists, stored into *pval,
+   FALSE if property does not exist.
+ */
+static int JS_TryGetProperty(JSContext *ctx, JSValue obj, JSAtom prop, JSValue *pval)
+{
+    /* could optimize by combining the calls, but hardly worth it */
+    int res = JS_HasProperty(ctx, obj, prop);
+    if (res < 0) {
+        *pval = JS_EXCEPTION;
+    } else
+    if (res == 0) {
```

Comment:
Not sure if it's the mobile interface, but it looks like this lacks an indentation level?

Reply:
I have simplified this function and further improved the integer array property getters.